internal/patch/patch.go

```diff
@@ -44,6 +44,15 @@ func NewValidator(patchReq string, s schema.Schema, extensions ...schema.Schema)
 		return OperationValidator{}, err
 	}
 
+	switch v := operation.Value.(type) {
+	// Okta also send the ID on PATCH requests.
+	// See: internal/idp_test/testdata/okta/update_group_name.json
+	// https://developer.okta.com/docs/reference/scim/scim-20/#update-a-specific-group-name
+	case map[string]interface{}:
+		delete(v, "id")
+		operation.Value = v
+	}
+
```

Comment:
Do we really need this special case? Can't we just ignore the mutability of patched attributes?

Reply:
We already ignore mutability. The problem is that `id` is not a part of a SCIM resource schema. It is assumed that every resource has an `id` and thus is omitted from the schema.
Since the validation of `path` and `value` happens based on these schemas, `id` will not be present in the schema, and thus the validation will fail.

Reply:
Does the RFC say something about patching common attributes? Maybe we should allow `externalId` and `meta` as well?

Reply:
We currently allow `externalId` to be patched.
But since `id` and `meta` are readOnly this should not be allowed.

Reply:
> But since `id` and `meta` are readOnly this should not be allowed.

Didn't you just mention we ignore mutability? 😅

Reply:
Yeah, we ignore mutability that is defined in the schema.
Yet `id` and `meta` are part of SCIM itself, and not implementation specific.

Reply:
> Yeah, we ignore mutability that is defined in the schema.
> Yet `id` and `meta` are part of SCIM itself, and not implementation specific.

I see, but is there a specific reason why we make this so complicated? IMO ignoring mutability everywhere would be much more consistent and understandable.

Reply:
Yes, the reason is that `id` and `meta` are added by the server. So they are not part of the `schema` package.
It is possible to add these before validating. Which makes it possible to target and change these with PATCH operation.

Should we also support this on filters then?